jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/RunConfigurationCredentialsExtension.kt

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.execution.CommonJavaRunConfigurationParameters
+import com.intellij.execution.RunConfigurationExtension
+import com.intellij.execution.configurations.JavaParameters
+import com.intellij.execution.configurations.RunConfigurationBase
+import com.intellij.execution.configurations.RunnerSettings
+import com.intellij.openapi.options.SettingsEditor
+import com.intellij.openapi.util.Key
+import com.intellij.ui.layout.panel
+import com.intellij.util.xmlb.XmlSerializer
+import org.jdom.Element
+import software.aws.toolkits.core.utils.tryOrNull
+import software.aws.toolkits.resources.message
+import javax.swing.JCheckBox
+import javax.swing.JComponent
+
+class RunConfigurationCredentialsExtension : RunConfigurationExtension() {
+    override fun isApplicableFor(configuration: RunConfigurationBase<*>): Boolean = configuration is CommonJavaRunConfigurationParameters
+
+    override fun <T : RunConfigurationBase<*>?> updateJavaParameters(configuration: T, params: JavaParameters, runnerSettings: RunnerSettings?) {
+        val project = configuration?.getProject() ?: return
+        val environment = params.env
+
+        if (configuration.getCopyableUserData(RUN_CONFIGURATION_CREDENTIALS_KEY)?.useCurrentConnection != true) {
+            return
+        }
+
+        tryOrNull {
+            project.activeCredentialProvider()
+        }?.resolveCredentials()?.toEnvironmentVariables()?.forEach { key, value -> environment.putIfAbsent(key, value) }
```

Comment:
This will put a session token even when access/secret is already added

I assume this will lead to calls failing

Reply:
ahhh good point.